public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -105,6 +105,7 @@ export class InspectJSONTab extends PureComponent<Props, State> {
           appEvents.emit(AppEvents.alertError, ['Unable to apply']);
         } else {
           const updates = JSON.parse(this.state.text);
+          dashboard!.shouldUpdateDashboardPanelFromJSON(updates, panel!);
```

Comment:
I'm not an expert in this area, but does `panel.refresh()` get invoked twice now that the dashboard grid is force updated?

Reply:
Hey @hugohaggmark :), that's a good point, I tested and double-checked the code, and it seems working fine(without being invoked twice). I think when we apply changes in the JSON  the `panel.refresh` is only called in `line 110 - InspectJSONTab`  and when we trigger the dashboard event and we force a re-render we are not calling it on the `DashboardGrid` component.

Reply:
👍